maven-enforcer-plugin/src/main/java/org/apache/maven/plugins/enforcer/EnforceMojo.java

```diff
@@ -127,11 +127,22 @@ public class EnforceMojo extends AbstractMojo {
     private PlexusConfiguration rules;
 
     /**
-     * List of strings that matches the EnforcerRules to execute.
+     * List of strings that matches the EnforcerRules to execute. Use <code>enforcer.rules</code> property instead.
+     *
+     * @deprecated
      */
     @Parameter(required = false, property = "rules")
     private List<String> commandLineRules;
 
+    /**
+     * List of strings that matches the EnforcerRules to execute. Replacement for the <code>rules</code> property.
+     * If both are specified, this one has a higher precedence.
+     *
+     * @since 3.2.0
+     */
+    @Parameter(required = false, property = "enforcer.rules")
+    private List<String> rulesToExecute;
+
```

Comment:
instead of adding next field we can try add `@Parameter` to setters method, so code can look like:

```java
@Parameter(required = false, property = "rules")
public void setCommandLineRules(String value) {
    this.rulesToExecute = value;
}

@Parameter(required = false, property = "enforcer .rules")
public void setRulesToExecute(String value) {
    this.rulesToExecute = value;
}
```

and we will have one field.

Reply:
I took a hybrid approach. I left the annotated field (so that we have the same setup as for the other fields) and added the setters to make sure the values are not getting overridden with empty lists (as the setter is getting called with empty list even if there is no configuration)